carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/BundleUtils.java

```diff
@@ -0,0 +1,157 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.maven.p2.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class composed of utility methods pertaining to a Bundles.
+ */
+public class BundleUtils {
+
+    private static final Pattern OSGI_VERSION_PATTERN = Pattern.compile("[0-9]+\\.[0-9]+\\.[0-9]+(\\.[0-9A-Za-z_-]+)?");
+    private static final Pattern ONLY_NUMBERS = Pattern.compile("[0-9]+");
+
+    /**
+     * Returns the OSGI version for the given artifact version.
+     *
+     * @param version artifact version
+     * @return OSGI version
+     */
+    public static String getOSGIVersion(String version) {
+        // This method is taken from org.apache.maven.shared.osgi.DefaultMaven2OsgiConverter to get the OSGI version
+        // from a give version artifact version.
+        String osgiVersion;
+
+        // Matcher m = P_VERSION.matcher(version);
+        // if (m.matches()) {
+        // osgiVersion = m.group(1) + "." + m.group(3);
+        // }
+
+        /* TODO need a regexp guru here */
```

Comment:
As I have commented here this method is taken from org.apache.maven.shared.osgi.DefaultMaven2OsgiConverter class. The TODO is in the original source. 

https://svn.apache.org/repos/asf/maven/shared/trunk/maven-osgi/src/main/java/org/apache/maven/shared/osgi/DefaultMaven2OsgiConverter.java